pkg/cloud/cloud_logging_transport.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package cloud
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/crlib/crtime"
+	"github.com/cockroachdb/redact"
+)
+
+// maybeAddLogging wraps the provided http.RoundTripper with a logging
+// transport if verbose logging is enabled.
+func maybeAddLogging(inner http.RoundTripper) http.RoundTripper {
+	if log.V(1) {
+		return &loggingTransport{inner: inner}
+	}
+	return inner
+}
+
+type loggingTransport struct {
+	inner http.RoundTripper
+}
+
+// RoundTrip implements http.RoundTripper.
+func (l *loggingTransport) RoundTrip(request *http.Request) (*http.Response, error) {
+
+	// NOTE: Body can be nil if the request has no body.
+	requestWatcher := &requestBodyTracker{inner: request.Body}
+	if request.Body != nil {
+		request = request.Clone(request.Context())
```

Comment:
This is subtle, it may be worth documenting that request.Clone() makes a shallow copy of body, so we avoid allocating a second body. The reader can also go read the docstring of Clone i guess. But also, feel free to just bors this.